job_creator.php

```diff
@@ -26,21 +35,16 @@ public function getInstallerVersion(
         string $installerBranchesJson = ''
     ): string
     {
-        $this->repoName = explode('/', $this->githubRepository)[1];
         // repo should not use installer
-        if (in_array($this->repoName, NO_INSTALLER_LOCKSTEPPED_REPOS) || in_array($this->repoName, NO_INSTALLER_UNLOCKSTEPPED_REPOS)) {
+        if (!$this->needsInstallerVersion()) {
```

Comment:
This change exits earlier under any scenario where we don't need installer for this repo.
Previously we had scenarios where we'd get through a bunch of this method only to hit a `return ''` - now it does most of the relevant checks upfront.

This is mostly unrelated refactoring, but made the rest of it a lot easier to debug and refactor the relevant stuff.